verification is broken

On current main (346f4862 ("Merge pull request #900 from peterfang/fix-unaligned-memmap")) `cargo verify` is broken:

```
$ cd kernel
$ cargo verify 
RUSTC="verus-rustc" RUSTUP_TOOLCHAIN="1.91.0" VERUS="/home/stefano/.cargo/bin/verus" VERUS_TARGETS="svsm,verify_external,verify_proof,verus_stub,vstd" "cargo" "build" "--features" "verus"
...
verification results:: 0 verified, 0 errors
verification results:: 0 verified, 0 errors
verification results:: 0 verified, 0 errors
error[E0425]: cannot find value `item_size` in this scope
    --> kernel/src/mm/alloc.rs:1034:28
     |
1034 |                 item_size: item_size as u64,
     |                            ^^^^^^^^^ not found in this scope
```

I tried this simple fix, but then others errors come up.
```diff
--- a/kernel/src/mm/alloc.rs
+++ b/kernel/src/mm/alloc.rs
@@ -1031,7 +1031,7 @@ impl HeapMemoryRegion {
             old(self).wf_next_pages(),
         ensures
             old(self).ens_allocate_pages_info(self, 0, PageInfo::Slab(SlabPageInfo {
-                item_size: item_size as u64,
+                item_size: N as u64,
             }) , ret, *perm),
     )]
     fn allocate_slab_page<const N: usize>(&mut self) -> Result<VirtAddr, AllocError> {
```

@ziqiaozhou @Chris-Hawblitzel can you take a look?

--- a/kernel/src/address.rs
+++ b/kernel/src/address.rs
@@ -527,7 +527,7 @@ impl<T> From<*mut T> for VirtAddr {
 
 #[verus_verify]
 impl<T> From<NonNull<T>> for VirtAddr {
-    #[verus_verify]
+    #[verus_verify(external_body)]
     #[inline]
     fn from(value: NonNull<T>) -> Self {
         Self::from(value.as_ptr())

--- a/kernel/src/mm/alloc.rs
+++ b/kernel/src/mm/alloc.rs
@@ -1029,9 +1029,10 @@ impl HeapMemoryRegion {
         with Tracked(perm): Tracked<&mut Option<UnitDeallocPerm>>
         requires
             old(self).wf_next_pages(),
+            N as u64 <= PageStorageType::SLAB_MASK,
         ensures
             old(self).ens_allocate_pages_info(self, 0, PageInfo::Slab(SlabPageInfo {
-                item_size: item_size as u64,
+                item_size: N as u64,
             }) , ret, *perm),
     )]
     fn allocate_slab_page<const N: usize>(&mut self) -> Result<VirtAddr, AllocError> {

--- a/verification/verify_external/src/lib.rs
+++ b/verification/verify_external/src/lib.rs
@@ -14,11 +14,14 @@
 #![no_std]
 #![allow(unused_braces)]
 #![allow(unexpected_cfgs)]
+#![cfg_attr(verus_keep_ghost, feature(sized_hierarchy))]
 
 // Add spec for convert traits
 pub mod convert;
 
 pub mod hw_spec;
+
+pub mod nonnull;
 
 pub mod ptr;
 

--- a/verification/verify_external/src/nonnull.rs
+++ b/verification/verify_external/src/nonnull.rs
@@ -1,0 +1,4 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+#[cfg(verus_keep_ghost)]
+include!("nonnull.verus.rs");

--- a/verification/verify_external/src/nonnull.verus.rs
+++ b/verification/verify_external/src/nonnull.verus.rs
@@ -1,0 +1,35 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+use core::ptr::NonNull;
+use vstd::prelude::*;
+use vstd::raw_ptr::*;
+
+verus! {
+
+/// External type specification for core::ptr::NonNull<T>
+///
+/// We use both external_type_specification and external_body because NonNull
+/// has private fields which are not supported for transparent datatypes.
+/// This pattern is used in vstd for similar types like ManuallyDrop.
+///
+/// We use accept_recursive_types to allow types like SlabPage that have
+/// Option<NonNull<Self>> fields to be verified.
+#[allow(missing_debug_implementations)]
+#[verifier::external_type_specification]
+#[verifier::external_body]
+#[verifier::accept_recursive_types(T)]
+pub struct ExNonNull<T: core::marker::PointeeSized>(NonNull<T>);
+
+/// Specification for NonNull::as_ptr
+///
+/// Returns the underlying raw pointer. The returned pointer is guaranteed
+/// to be non-null (addr != 0).
+pub assume_specification<T: core::marker::PointeeSized>[ NonNull::<T>::as_ptr ](
+    self_: NonNull<T>,
+) -> (ret: *mut T)
+    ensures
+        ret@.addr != 0,
+    opens_invariants none
+    no_unwind
+;
+
+} // verus!

